[T10-3] LearnVocabulary

russellry opened the pull request:
Travis Build is still failing - even before coding, needs a little more attention.

Code wise:
Changes implemented - dictionary function, add/find, grouping and trivia

Comments:
Dictionary has been fully worked out, just needs to be integrated, will probably need a lot of debugging.
Add/find, grouping and trivia has a few bugs

Also, we are aware of some discrepencies in the naming of our document and will address that by next week! We will also update the author's name as well.

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -1,966 +1,146 @@
-. Go to `File` > `Settings...` (Windows/Linux), or `IntelliJ IDEA` > `Preferences...` (macOS)
-. Select `Editor` > `Code Style` > `Java`
-. Click on the `Imports` tab to set the order
+System checks if word exists in the dictionary and displays if the word could be successfully added or not. It also displays the meaning of the word.
```
Too many steps in this one line. Ideally it should be split up as the MSS and the exception workflow. Something like:
MSS:
1. User enters a word to add to the review list.
2. System adds word to the dictionary.
3. System prompts user with word successfully added message
4. System displays newly added word's meaning
Use case ends

Extensions:
2a. Word already exists in the dictionary
2a1. System prompts user with word already exists message
Use case ends

russellry replied:
@alexehrs @Harryqu123 @danielkoh94

parvathy-sudhir-pillai reviewed (commented):
Overall:
Good job on meeting all requirements for the v1.1 team component. However, some areas need more work. Hope you can tackle those areas in upcoming features.

Some detailed feedback:

* ReadMe adoc
1. Product description still refers to AB-4's functionality (point 2). 
2. Describe what LearnV is in the homepage
3. Remove top SE EDU nav bar and link to LOs

* View on GitHub link
Results in 404. Check that you have correctly followed the steps for auto-publishing of docs

* Contact Us adoc
Details have to be updated

* About Us adoc
1. Photos aren't visible. Please update ASAP.

* Developer Guide adoc
1. Looks like the auto publish is not properly enabled. The page is all cluttered up and not being displayed in the correct manner.
2. Two groups of Group User stories provided. Use either one, preferably the one grouped by priority
3. Use case steps are too complex and not atomic. Suggest splitting up into multiple steps as in the comment provided.

russellry commented:
Noted:
Ui.png has been updated (❗️ try to do by next milestone)
Milestones created: v1.2 [Please ensure milestones are setup until 'v1.4'] (❗️ try to do by next milestone)
Issues marked as type.Story e.g., (❗️ try to do by next milestone)
Issues allocated to v1.3 e.g., (❗️ try to do by next milestone)
Build passing (❗️ try to do by next milestone)

Individual Progress:
Gave review comments to other PRs (❗️ try to do by next milestone) - What is this exactly? On this PR? Or on other people's PR?

parvathy-sudhir-pillai reviewed (commented):
### Feedback for v1.2

**Read Me ascii doc**
* The Introduction section refers to a Section 2, “Quick Start” which is not part of Read Me, but in User guide. Update description
![t10_1](https://user-images.githubusercontent.com/9066342/47256954-4582b680-d4ba-11e8-9227-6346a5043a1c.png)

**User guide ascii doc**
* Use better descriptions for what each command does. Some of them are not very clear. For e.g. Group words. The description reads group words accordingly. According to what?
![t10_2](https://user-images.githubusercontent.com/9066342/47256974-91cdf680-d4ba-11e8-872b-744beb9f26ad.png)

**Developer guide ascii doc**
* If a feature is not implemented, you can mark it as coming in v2.0
* User stories are repeated twice. Use the version in the Appendix as they are prioritised.
![t10_3](https://user-images.githubusercontent.com/9066342/47257062-77484d00-d4bb-11e8-9b8b-d1e0d82b9678.png)
![t10_4](https://user-images.githubusercontent.com/9066342/47257072-921ac180-d4bb-11e8-8530-861b52ac5ea3.png)


### Overall
* Good effort. Work on getting the product to a user-testable state by v1.3
* Catch up on documentation work. Good job adding some design descriptions, but not enough new UML diagrams to give feedback on your UML modeling skills. Try to add more soon.

danielkoh94 committed:
Added JUnit test for trivia and triviaAns

danielkoh94 committed:
Added EOF

danielkoh94 committed:
Added and updated html page in browser panel

danielkoh94 committed:
Fixed test cases

danielkoh94 committed:
Fixed test cases

danielkoh94 committed:
Merge pull request #94 from danielkoh94/master

Added tests for trivia command and updates to browser panel.

Commit:
Fix bugs for test of not exist tag

Commit:
change back mainapp

Commit:
showGroup Added

Commit:
Further feature for showgroup --- check all existed group

Commit:
test for cravis

alexehrs committed:
Update User Guide with Word Of The Day description

danielkoh94 committed:
Testing stuff

danielkoh94 committed:
Test

danielkoh94 committed:
Set theme jekyll-theme-minimal

alexehrs committed:
Add Dev guide description on word of the day command + small fixes on other things in Dev guide

alexehrs committed:
Fix typos

alexehrs committed:
Merge pull request #98 from alexehrs/master

Add description on Word of The Day in Developer guide.

danielkoh94 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

danielkoh94 committed:
Merge branch 'master' of https://github.com/danielkoh94/main

danielkoh94 committed:
Changed theme

danielkoh94 committed:
Added newline

danielkoh94 committed:
Deleted config.yml

danielkoh94 committed:
deleted file

danielkoh94 committed:
Merge pull request #99 from danielkoh94/master

Added html file git pages

Harryqu123 committed:
Merge pull request #97 from Harryqu123/master

ShowGroup Feature

danielkoh94 committed:
Moved html pages to docs

danielkoh94 committed:
Fixed typo

danielkoh94 committed:
Merge pull request #100 from danielkoh94/master

Moved html pages to docs

danielkoh94 committed:
Changed url for browser panel

danielkoh94 committed:
Fixed assert commands

danielkoh94 committed:
Removed unused imports

danielkoh94 committed:
Fixed BrowserPanelTest.java

danielkoh94 committed:
Merge pull request #101 from danielkoh94/master

Fixed test cases

russellry committed:
Merge branch 'master' into master

russellry committed:
Merge pull request #90 from RussellOng95/master

updated icon for learnvocabulary, removed address_book_32.png

russellry committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

# Conflicts:
#	config.json

russellry committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

russellry committed:
Checkstyle fixes

russellry committed:
Merge pull request #102 from RussellOng95/master

 Checkstyle fixes

danielkoh94 committed:
Bug fixes for add, edit and browser panel

danielkoh94 committed:
Fixed test cases

danielkoh94 committed:
Fixed test cases. Add command does not allow multiple names or meaning

Commit:
Manually Indicate code owner

Harryqu123 committed:
Merge pull request #103 from Harryqu123/master

Manually Indicate code owner v1.1

Commit:
Owner Fix v1.2 and 1.3 without test by now

Harryqu123 committed:
Merge branch 'master' into master

Harryqu123 committed:
Merge pull request #104 from Harryqu123/master

Owner Fix v1.2 and 1.3 without test by now

Commit:
bugs for groupadd and clear

Commit:
Merge branch 'master' of https://github.com/Harryqu123/main

danielkoh94 committed:
Merge pull request #105 from danielkoh94/master

Fix some bugs

Harryqu123 committed:
Merge pull request #106 from Harryqu123/master

Bug for groupadd and clear

russellry committed:
commit locally (learnvocab, model, model manager)

russellry committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

alexehrs committed:
Update UserGuide.adoc

Commit:
fix checkstyle

Harryqu123 committed:
Merge pull request #107 from Harryqu123/master

fix checkstyle

Commit:
fix edit bug and toLearn bug

Harryqu123 committed:
Merge pull request #108 from Harryqu123/master

fix edit bug and toLearn bug

Commit:
CRLF---LF

Harryqu123 committed:
Merge pull request #109 from Harryqu123/master

CRLF---LF

russellry committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

Commit:
TEST

Harryqu123 committed:
Merge pull request #111 from Harryqu123/master

crlf-lf

Commit:
config_test

Harryqu123 committed:
Merge pull request #112 from Harryqu123/master

config_test

alexehrs committed:
fix word of the day error mesage bug

alexehrs committed:
Merge pull request #113 from alexehrs/master

fix word of the day error mesage bug

Commit:
fix testcases

Harryqu123 committed:
Merge pull request #114 from Harryqu123/master

fix testcases

Commit:
fix bug

Harryqu123 committed:
Merge pull request #115 from Harryqu123/master

fix testcase checkstyle

Commit:
showgroupcommand testcases

Commit:
fix

Harryqu123 committed:
Merge pull request #117 from Harryqu123/master

showgroupcommand test cases

Commit:
no message

Harryqu123 committed:
Merge pull request #118 from Harryqu123/master

no message

Commit:
groupaddcommand test case

Commit:
groupadd command test case

Commit:
groupadd command owner

Commit:
LF?

Harryqu123 committed:
Merge pull request #121 from Harryqu123/master

groupaddcommand test cases

Commit:
fix a groupdelete bug and groupdelete test cases

Commit:
crlf fix

Harryqu123 committed:
Merge pull request #122 from Harryqu123/master

groupdelete test cases

Commit:
tagpredicate testcases

Harryqu123 committed:
Merge pull request #123 from Harryqu123/master

tagpredicate testcases

danielkoh94 committed:
Edited and formatted user guide.

danielkoh94 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

Commit:
groupadd parser testcase

Harryqu123 committed:
Merge pull request #124 from Harryqu123/master

groupadd parser testcase

Commit:
groupadd fix

Harryqu123 committed:
Merge pull request #125 from Harryqu123/master

groupadd fix

Commit:
group delete parse test cases

Harryqu123 committed:
Merge pull request #126 from Harryqu123/master

group delete parse test cases

Commit:
groupdelete fix

Harryqu123 committed:
Merge pull request #127 from Harryqu123/master

groupdelete fix

danielkoh94 committed:
Update user guide

danielkoh94 committed:
Merge pull request #128 from danielkoh94/master

Updated user guide

Harryqu123 committed:
Update UserGuide.adoc

russellry committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

Harryqu123 committed:
Update UserGuide.adoc

russellry committed:
Changes to remove Ui for TagPanel

Harryqu123 committed:
Update UserGuide.adoc

Harryqu123 committed:
Update UserGuide.adoc

russellry committed:
v1.3 - compilation plus dev and user guide

russellry committed:
Learn Command test cases + general fixes

russellry committed:
Merge branch 'master' into master

russellry committed:
Checkstyle fix

russellry committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

# Conflicts:
#	docs/UserGuide.adoc

russellry committed:
Merge branch 'master' of https://github.com/RussellOng95/main

russellry committed:
Merge pull request #129 from RussellOng95/master

Merge testing for v1.3

alexehrs committed:
Update user guide with word of the day

russellry committed:
fixed issue #133

russellry committed:
Merge pull request #144 from RussellOng95/master

Bug fixes #133

russellry committed:
test cases for Dictionary fixes

russellry committed:
fix #150 handle illegal show command.

russellry committed:
Updated user guide for show and learn commands, fixed #145 as well

russellry committed:
bug fix #142

russellry committed:
fixed edit command allowing for empty tags & fix test cases

russellry committed:
Merge pull request #151 from RussellOng95/master

test cases for Dictionary fixes

alexehrs committed:
create wordofthedaytest.java

alexehrs committed:
add test for WordOfTheDayCommand

alexehrs committed:
add wordOfTheDayParserTest.java

alexehrs committed:
add test for word of the day parser

alexehrs committed:
fix checkstyle errors

alexehrs committed:
Merge pull request #154 from alexehrs/master

Add tests for wordofthedaycommand and parser

danielkoh94 committed:
Updated answer command to be case-insensitive

danielkoh94 committed:
Merge pull request #155 from danielkoh94/master

Updated answer command to be case-insensitive

danielkoh94 committed:
Update UserGuide.adoc

danielkoh94 committed:
Update UserGuide.adoc

russellry committed:
Update Developer Guide

Need to update dead links

russellry committed:
Update formatting and design section

Everybody to edit design section

russellry committed:
Formatting finish

russellry committed:
Update DeveloperGuide.adoc

russellry committed:
Update DeveloperGuide.adoc

russellry committed:
Update all addressbook to learnvocabulary

russellry committed:
Delete LogicClassDiagram.png

russellry committed:
Added new LogicClassDiagram.png

russellry committed:
Update #1 for developer guide for diagrams

russellry committed:
Changed diagrams to fit better

russellry committed:
Final edits to diagrams for copied version for undo

russellry committed:
Updating developerguide for learn command

to be edited

russellry committed:
Merge pull request #157 from RussellOng95/master

Update format of Developer Guide

russellry committed:
Added learn sequence diagram

russellry committed:
Merge branch 'master' of https://github.com/RussellOng95/main

russellry committed:
Finalized Learn & Basic Features in developer guide

russellry committed:
Updated user guide for edit, add, learn and show commands

russellry committed:
Merge pull request #158 from RussellOng95/master

Updated User and Developer Guide

Harryqu123 committed:
Update DeveloperGuide.adoc

Harryqu123 committed:
Update DeveloperGuide.adoc

Harryqu123 committed:
Update DeveloperGuide.adoc

Harryqu123 committed:
Update DeveloperGuide.adoc

danielkoh94 committed:
Updated Developer Guide

danielkoh94 committed:
Updated developer guide

danielkoh94 committed:
Merge pull request #159 from danielkoh94/master

Updated trivia section in developer guide

Harryqu123 committed:
Add files via upload

Harryqu123 committed:
Update DeveloperGuide.adoc

Harryqu123 committed:
Add files via upload

Harryqu123 committed:
Update DeveloperGuide.adoc

danielkoh94 committed:
Added PPP for danielkoh94, minor changes in developer's guide and user's guide

danielkoh94 committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

danielkoh94 committed:
Fixed trailing whitespace

danielkoh94 committed:
Merge pull request #160 from danielkoh94/master

Added PPP for danielkoh94

russellry committed:
Update User Guide with proper formatting

@danielkoh94 @Harryqu123 @alexehrs please add in your relevant commands (Under Command Summary) for the last section!

russellry committed:
Update Developer Guide formatting

russellry committed:
Finalize Developer Guide Formatting

russellry committed:
Removed redundant sections in appendix

russellry committed:
Update Learn command developer guide

russellry committed:
Added tags for PPP (Russell)

russellry committed:
Changed placement of future features

russellry committed:
Remove trailing whitespace

russellry committed:
Create PPP for Russell

russellry committed:
Remove trailing white spaces

russellry committed:
Update Responsibilities and PPP

russellry committed:
Refactored "toLearn" tag as a static variable in Tag class

russellry committed:
Added additional test cases for LearnCommandParser

russellry committed:
Changed default tags to static Tag class tags

russellry committed:
Refactored parsing definition into its own method

russellry committed:
Changed Learn Sequence Diagram - added new method

russellry committed:
Cleanup code

russellry committed:
Merge pull request #161 from RussellOng95/master

Refactor Code, add test cases

Harryqu123 committed:
Create harryqu123.adoc

Harryqu123 committed:
Update UserGuide.adoc

Harryqu123 committed:
Update harryqu123.adoc

Harryqu123 committed:
Update AboutUs.adoc

alexehrs committed:
add word of the day sequential diagram plus text changes

alexehrs committed:
fixed errors + add text to user guide

alexehrs committed:
fix small error in user guide

alexehrs committed:
add word_OD to summary

alexehrs committed:
fix typo

alexehrs committed:
Merge pull request #162 from alexehrs/master

Add text and sequence diagram for Word Of The Day into Dev and User guides

parvathy-sudhir-pillai reviewed (commented):
* It is better to have only one email to contact in the Contact Us page than to list out the mail ids of all team members.
* All sequence diagrams except the group feature shows interaction between multiple components. For e.g. in the trivia feature, the low-level interactions between Logic and Model is shown. While in few diagrams, for e.g. Word of the day feature, there is no clear demarcation as to which component the classes depicted belong to. Reiterating what was mentioned during the tutorial. CS2103 encourages multi-level sequence diagrams. Higher level interactions between components (e.g. Logic and Model) in a separate diagram and lower level interactions between sub-components (e.g. Parser and Command) within a component (Logic) in a separate diagram. The lower level diagram should not include interactions between components (e.g. Logic and Model).
* Try to follow the example diagram of the group feature in your developer guide for other features as well.

**Good job, Team!
All the best for your v1.4 submission and presentation!**

alexehrs committed:
add PPP for alexehrs part 1

alexehrs committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T10-3/main

alexehrs committed:
add PPP for alexehrs part 2

alexehrs committed:
fix typos and further enchance

alexehrs committed:
fix diagrams for word of the day

alexehrs committed:
add portfolio link to alexehrs

alexehrs committed:
Merge pull request #163 from alexehrs/master

Add PPP for alexehrs

parvathy-sudhir-pillai reviewed (dismissed):
**Good job, Team!
All the best for your v1.4 submission and presentation!**

danielkoh94 committed:
Changes to trivia and triviaAns sequence diagram. Minor changes to user guide and danielkoh PPP

danielkoh94 committed:
Update danielkoh94.adoc

danielkoh94 committed:
Update danielkoh94.adoc

Harryqu123 committed:
Merge pull request #1 from CS2103-AY1819S1-T10-3/master

update

Commit:
bug fix for showgroup and groupdelete_undo

Harryqu123 committed:
Merge pull request #165 from Harryqu123/master

fix undo_groupdelete

Commit:
no message

Harryqu123 committed:
Merge pull request #166 from Harryqu123/master

new line config

Commit:
groupdelete with no para and test fix

Harryqu123 committed:
Merge pull request #167 from Harryqu123/master

groupdelete with no para and test fix

Commit:
checkstyle fix and showgroup fix

Harryqu123 committed:
Merge pull request #168 from Harryqu123/master

checkstyle fix and showgroup fix

Commit:
groupadd invalid name

danielkoh94 committed:
Merge pull request #164 from danielkoh94/master

Changes to trivia sequence diagrams

alexehrs committed:
new use case about word of the day

alexehrs committed:
Merge pull request #170 from alexehrs/master

new use case about word of the day

Commit:
no message

Harryqu123 committed:
Merge pull request #171 from Harryqu123/master

test case

russellry committed:
Updated Manual testing for learn, show, edit, add.

russellry committed:
Remove trailing white spaces

russellry committed:
Updated LV to LearnVocuabulary

russellry committed:
Updated Introduction

russellry committed:
Update formatting

russellry committed:
Delete johndoe.adoc

russellry committed:
Remove trailing whitespace

danielkoh94 committed:
Update DeveloperGuide.adoc

Added trivia testing instructions

russellry committed:
Updated Edit and Add command instructions

russellry committed:
Updated User Guide (edit and add)

russellry committed:
Update formatting

russellry committed:
Update formatting

Commit:
no message

Harryqu123 committed:
Merge pull request #172 from Harryqu123/master

no message

russellry committed:
Update formatting

Commit:
no message

Commit:
no message

Harryqu123 committed:
Merge pull request #173 from Harryqu123/master

typo

russellry committed:
Update formatting

Harryqu123 committed:
Update DeveloperGuide.adoc

Commit:
no message

Harryqu123 committed:
Merge pull request #174 from Harryqu123/master

no message

Harryqu123 committed:
Update DeveloperGuide.adoc

russellry committed:
Update format of Developer Guide

russellry committed:
Update User Guide 

(replaced basic tag with group tag) - for @Harryqu123

russellry committed:
Update Developer Guide

russellry committed:
Update Developer Guide

russellry committed:
Update Developer Guide

alexehrs committed:
fix bugs in alexehrs PPP

alexehrs committed:
Merge pull request #175 from alexehrs/master

fix bugs in alexehrs PPP

alexehrs committed:
fixes

alexehrs committed:
fix tags for PPP

alexehrs committed:
Merge pull request #176 from alexehrs/master

Bug fixes alexehrs PPP
